public/app/tasks/task-list.controller.js

```diff
@@ -17,26 +17,27 @@
         self.newTask = {};
         self.initializing = true;
 
-        TaskService.getList().then(function (data) {
-            self.selectedId = $stateParams.id;
-            self.searchIndex = _initSearchIndex(data);
-            if ($stateParams.search) {
-                _initList(_initSearch(data));
-            }
-            else {
-                _initList(data);
-            }
-        });
+        _initList();
 
         function _initSearch(data) {
             var foundIds = self.searchIndex.search($stateParams.search)
```

Comment:
I miss the sense of the row. You search through the search index to find what values should be put to the search index?